src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -334,6 +337,11 @@ CompleteShardPlacementTransactions(XactEvent event, void *arg)
 	shardConnectionHash = NULL;
 	XactModificationLevel = XACT_MODIFICATION_NONE;
 	subXactAbortAttempted = false;
+
+	if (MultiShardCommitProtocol == COMMIT_PROTOCOL_BARE)
+	{
+		MultiShardCommitProtocol = SavedMultiShardCommitProtocol;
+	}
```

Comment:
> Now you can argue that that's unlikely to be a problem here, but I still think it's bad.

I kind of think it's _extremely_ unlikely to be a problem? Someone would have to somehow inspect the GUC values during a `VACUUM`, which resets the value at the end. Maybe my cold or flu or whatever is preventing me from seeing a more inventive solution, but is this really an issue?

If you think it is (even perhaps if you think the problem is unrealistic), just give me a veto and I'll switch to the GUC approach.

And I would need to prevent users from setting that value, right?